src/components/instantlaunches/index.js

```diff
@@ -0,0 +1,329 @@
+        })
+        .then((submission) => submitAnalysis(submission)) // submit the analysis
+        .then((analysisResp) => getAnalysis(analysisResp.id));
+};
+
+const useStyles = makeStyles((theme) => ({
+    progress: {
+        marginTop: theme.spacing(2),
+        marginBottom: theme.spacing(5),
+    },
+    inProgress: {
+        marginBottom: theme.spacing(3),
+    },
+    closeAuto: {
+        marginBottom: theme.spacing(3),
+    },
+}));
+
+/**
+ * The loading dialog that displays while the acutal submission call is in progress.
+ *
+ * @param {Object} props - The component props.
+ * @param {boolean} props.open - Whether or not the dialog is open.
+ */
+export const InstantLaunchSubmissionDialog = ({ open }) => {
+    const baseID = buildID(ids.BASE, ids.SUBMISSION, ids.DIALOG);
+    const { t } = useTranslation("instantlaunches");
+    const classes = useStyles();
+
+    return (
+        <Dialog
+            open={open}
+            onClose={(event) => {
+                event.stopPropagation();
+                event.preventDefault();
+            }}
+            id={baseID}
+        >
+            <DialogTitle id={buildID(baseID, ids.TITLE)}>
+                {t("submittingInstantLaunch")}
+            </DialogTitle>
+
+            <DialogContent>
+                <div
+                    className={classes.progress}
+                    id={buildID(baseID, ids.PROGRESS)}
+                >
+                    <LinearProgress />
+                </div>
+
+                <Typography variant="h6" className={classes.inProgress}>
+                    {t("submissionInProgress")}
+                </Typography>
+
+                <Typography variant="body2" className={classes.closeAuto}>
+                    {t("dialogWillCloseAutomatically")}
+                </Typography>
+            </DialogContent>
+        </Dialog>
+    );
+};
+
+/**
+ * An IconButton with a handler that launches the app in the instant launch with
+ * the resource as an input.
+ *
+ * @param {Object} props
+ * @param {Object} props.instantLaunch - The instant launch to use.
+ * @param {Object} props.resource - The resource to use as an input to the instant launch.
+ */
+const InstantLaunchButton = ({
+    instantLaunch,
+    resource = {},
+    showErrorAnnouncer,
+    size = "medium",
+    themeSpacing = 0,
+    color = "primary",
+}) => {
+    const baseID = buildID(ids.BASE, ids.LAUNCH, ids.BUTTON);
+    const router = useRouter();
+    const [open, setOpen] = React.useState(false);
+    const theme = useTheme();
+
+    return (
+        <IconButton
+            id={baseID}
+            variant="contained"
+            size={size}
+            style={{ marginLeft: theme.spacing(themeSpacing) }}
+            color={color}
+            onClick={async (event) => {
+                event.stopPropagation();
+                event.preventDefault();
+
+                setOpen(true);
+
+                await instantlyLaunch(instantLaunch, resource)
```

Comment:
I tried this, but it ended up locking up submission calls. I'm going to leave it alone for now so I don't spend a ton of time churning on this, since we're on a time limit.